utam-core/src/main/java/utam/core/framework/element/NavigationImpl.java

```diff
@@ -98,4 +105,59 @@ public void closeWindow(String targetUrl) {
   public void closeWindow() {
     this.driverAdapter.close();
   }
+
+  @Override
+  public int getWindowCount() {
+    return this.driverAdapter.getWindowHandles().size();
+  }
+
+  @Override
+  public Window currentWindow() {
+    return new WindowImpl(this.factory);
+  }
+
+  @Override
+  public void setupWaitForNewWindow() {
+    oldWindowCount = getWindowCount();
+    oldWindowHandles = new HashSet<>(this.driverAdapter.getWindowHandles());
+
+    this.setupForNewWindow = true;
+  }
+
+  @Override
+  public Window waitForNewWindow() {
+    if (!this.setupForNewWindow) {
+      throw new UtamError(ERR_SETUP_NOT_RUN);
+    }
+
+    // Wait for the window count to increase indicating a new window
+    this.driverAdapter.waitFor(() -> getWindowCount() > oldWindowCount, "wait for a new window to open", null);
+
+    // The difference between the sets of old and new window handles should be the new windows handle
+    // Note: if more than one window opens this method will return first that appears in the list of window handles
+    Set<String> currentWindowHandles = new HashSet<>(this.driverAdapter.getWindowHandles());
+    currentWindowHandles.removeAll(oldWindowHandles);
+    String newWindowsHandle = currentWindowHandles.toArray(new String[1])[0];
+
+    oldWindowCount = -1;
+    oldWindowHandles = null;
+    this.setupForNewWindow = false;
+
+    this.driverAdapter.switchTo(newWindowsHandle);
+    return new WindowImpl(this.factory);
+  }
+
+  @Override
+  public PageObject waitForNewWindowAndLoad(Class<? extends RootPageObject> pageObjectType) {
+    waitForNewWindow();
+
+    return factory.getPageContext().getBean(pageObjectType);
```

Comment:
Instead of `getPageContext()`, what you want here is:

    T instance = factory.create(type);
    instance.load();
    return instance;